q2_types/genome_data/_methods.py

```diff
@@ -13,32 +13,59 @@
 from qiime2.util import duplicate
 
 from q2_types.genome_data import (SeedOrthologDirFmt, OrthologAnnotationDirFmt,
-                                  LociDirectoryFormat)
+                                  LociDirectoryFormat, GenesDirectoryFormat,
+                                  ProteinsDirectoryFormat)
 
 
 def collate_loci(loci: LociDirectoryFormat) -> LociDirectoryFormat:
-    """
-    Collate the individual loci directories from the partitions.
-    Parameters:
-    - loci: A list of LociDirectoryFormat containing the gff files.
-    Returns:
-    - collated_loci: A LociDirectoryFormat object containing the
-    collated gff files.
-    """
-    collated_loci = LociDirectoryFormat()
-    for loci_dir in loci:
-        for fp in loci_dir.path.iterdir():
-            try:
-                duplicate(
-                    fp,
-                    collated_loci.path / os.path.basename(fp)
-                )
-            except FileExistsError:
-                warnings.warn(
-                    f"Skipping {fp}. File already exists "
-                    f"in the destination directory."
+    return _collate_helper(dir_fmts=loci, collated=LociDirectoryFormat())
+
+
+def collate_ortholog_annotations(
+    ortholog_annotations: OrthologAnnotationDirFmt
+) -> OrthologAnnotationDirFmt:
+    return _collate_helper(
+        dir_fmts=ortholog_annotations,
+        collated=OrthologAnnotationDirFmt())
+
+
+def collate_genes(genes: GenesDirectoryFormat) -> (
+        GenesDirectoryFormat):
+    return _collate_helper(
+        dir_fmts=genes,
+        collated=GenesDirectoryFormat())
+
+
+def collate_proteins(proteins: ProteinsDirectoryFormat) -> (
+        ProteinsDirectoryFormat):
+    return _collate_helper(
+        dir_fmts=proteins,
+        collated=ProteinsDirectoryFormat())
+
+
+def _duplicate_warning(src, dst):
```

Comment:
This function is an overkill - all it does is adding a try-except. Just introduce the block in the _collate_helper directly.

Reply:
Agree to disagree. But fine. At least change the name of this function because it sounds like you're duplicating the warnings.